WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -342,9 +345,11 @@ def post_results
           body += "#{record_strs.join(", ")}.  \n" # Trailing spaces for markdown give us a <br>
         end
       end
-      comp.update!(results_posted_at: Time.now)
-      comp.competitor_users.each { |user| user.notify_of_results_posted(comp) }
-      comp.registrations.accepted.each { |registration| registration.user.notify_of_id_claim_possibility(comp) }
+      unless comp.results_posted?
```

Comment:
By the time we get here, won't the results always be *not* posted (you check earlier if they are posted and bailed out)